### PR TITLE
git.py: Allow setting of submodule push branch

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
--push is used to set the branch in `.gitmodules`.

### CODE REVIEW

**Comment by rleigh-codelibre** [comment]
Green in https://ci.openmicroscopy.org/view/Mgmt/job/SCC-merge/957/console

File: scc/git.py [hunk 1]
```diff
@@ -1007,7 +1007,8 @@ def find_candidate_branches(self, filters,
 
 class GitRepository(object):
 
-    def __init__(self, gh, path, remote="origin", repository_config=None):
+    def __init__(self, gh, path, remote="origin", push_branch="origin",
```
Comment (sbesson) [inline scc/git.py]: In the case where `--push-branch` is not passed to the original command, am I correct that with the current implementation `scc merge <xxx> --update-gitmodules` would unconditionally write `branch = origin` in the `.gitmodules`?

My expectation would be that in that case, we might not want to modify this key i.e. set the default here to `None` and conditionally skip the setting of `submodule.<name>.branch`.

Comment (rleigh-codelibre) [reply scc/git.py]: Please see the additional commit.

**Comment by rleigh-codelibre** [comment]
Green in https://ci.openmicroscopy.org/view/Mgmt/job/SCC-merge/958/console

**Comment by sbesson** [comment]
Functionally tested using the sandbox repository. Without `--push` arguments

```sh
(venv) sbesson@ls30630:snoopys-sandbox (dev_4_4) $ git reset --hard origin/master && git submodule update  --init
warning: unable to rmdir 'snoopys-sandbox-2': Directory not empty
HEAD is now at 7477b4d Initial commit
(venv) sbesson@ls30630:snoopys-sandbox (dev_4_4) $ python /opt/ome/snoopycrimecop/scc/main.py merge dev_4_4  --update-gitmodules --push test
2018-02-14 13:31:35,488 [   scc.merge] INFO  Merging Pull Request(s) based on dev_4_4
....
(venv) sbesson@ls30630:snoopys-sandbox (dev_4_4) $ git diff HEAD^
diff --git a/snoopys-sandbox-2 b/snoopys-sandbox-2
index 33abb18..ac7b359 160000
--- a/snoopys-sandbox-2
+++ b/snoopys-sandbox-2
@@ -1 +1 @@
-Subproject commit 33abb18f09519d2174a86e1933a6d62dc959f2f9
+Subproject commit ac7b359d097ca1c038f17e2508757f3b256836d9
```

With `--push` arguments

```
(venv) sbesson@ls30630:snoopys-sandbox ((aed9f2f...)) $ python /opt/ome/snoopycrimecop/scc/main.py merge dev_4_4  --update-gitmodules --push test
2018-02-14 13:57:35,872 [   scc.merge] INFO  Merging Pull Request(s) based on dev_4_4
...
2018-02-14 13:57:56,995 [   scc.merge] INFO  Merged branch pushed to https://github.com/sbesson/snoopys-sandbox/tree/test
(venv) sbesson@ls30630:snoopys-sandbox ((2ca119a...)) $ git diff HEAD^
diff --git a/.gitmodules b/.gitmodules
index b0dae12..9a6e374 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,4 @@
 [submodule "snoopys-sandbox-2"]
        path = snoopys-sandbox-2
        url = git://github.com/openmicroscopy/snoopys-sandbox-2
+       branch = test
diff --git a/snoopys-sandbox-2 b/snoopys-sandbox-2
index 33abb18..53b6588 160000
--- a/snoopys-sandbox-2
+++ b/snoopys-sandbox-2
@@ -1 +1 @@
-Subproject commit 33abb18f09519d2174a86e1933a6d62dc959f2f9
+Subproject commit 53b65889596635873a10ef0060a4739cce6860af
```

Using my local `scc` version (0.8.0), both commands result in the first behavior. This

**Comment by joshmoore** [comment]
Reading that as good to go for 0.9.2. Merging & releasing.

**Pull request merged**